ui/src/app/views/pipeline/show/workflow/pipeline.workflow.html

```diff
@@ -1,33 +1,71 @@
+                        [class.five]="pipeline.stages.length === 4"
+                        [class.six]="pipeline.stages.length === 5"
+                        [class.seven]="pipeline.stages.length === 6">
+                        <div class="stageItem">
+                            <a class="pointing" (click)="openEditModal(stage)"><i class="setting icon"></i>{{stage.name}}</a>
+                            <i class="move icon"></i>
+                            <ul>
+                                <li *ngFor="let j of stage.jobs; let i = index" id="Job{{i}}">
+                                    <div class="job ui segment pointing" [class.inactive]="!j.enabled" (click)="selectJob(j, stage)"
+                                         [class.active]="j.pipeline_action_id === selectedJob?.pipeline_action_id">
+                                        {{j.action.name}}
+                                    </div>
+                                </li>
+                                <li class="new job">
+                                    <div class="new job ui segment pointing" (click)="addJob(stage)">
+                                        {{ 'step_add_job' | translate }}
+                                    </div>
+                                </li>
+                            </ul>
+                        </div>
+                    </li>
+                    <li class="stage"
+                        [class.one]="pipeline.stages.length === 0"
+                        [class.two]="pipeline.stages.length === 1"
+                        [class.three]="pipeline.stages.length === 2"
+                        [class.four]="pipeline.stages.length === 3"
+                        [class.five]="pipeline.stages.length === 4"
+                        [class.six]="pipeline.stages.length === 5"
+                        [class.seven]="pipeline.stages.length === 6">
+                        <div class="new stageItem">
+                            <ul>
+                                <li class="job">
+                                    <div id="AddStageAndJob" class="new job ui segment pointing" (click)="addStageAndJob()">
+                                        {{ 'step_add_job' | translate }}
+                                    </div>
+                                </li>
+                            </ul>
+                        </div>
+                    </li>
+                </ul>
+            </div>
+        </div>
     </div>
-  <div class="step" *ngIf="pipeline.permission === 7">
-    <div class="content">
-      <div class="title">
-        <a class="pointing" (click)="addStage()" id="AddStageLink">
-          <i class="green plus circle icon"></i>{{ 'stage_add_label' | translate }}
-        </a>
-      </div>
+    <div *ngIf="selectedJob">
+        <h3></h3>
+        <app-action [project]="project" [suggest]="suggest" [action]="selectedJob.action" [edit]="pipeline.permission === 7"
```

Comment:
I think it could be more readable to set permission to an enum in model in order to know which permission is 7